src/prefect/cli/deployment.py

```diff
@@ -1024,7 +1024,11 @@ async def build(
     infra_overrides = {}
     for override in overrides or []:
         key, value = override.split("=", 1)
-        infra_overrides[key] = value
+        override_key, sub_key = key.split(".", 1)
+        if override_key in infra_overrides:
+            infra_overrides[override_key].update({sub_key: value})
+        else:
+            infra_overrides[override_key] = {sub_key: value}
```

Comment:
This is awesome, but we have the opportunity to handle arbitrary nesting of overrides. This will handle one level deep, but we have a utility called `flatdict_to_dict` that can help us handle arbitrary depth. 

This code should allow us to accept overrides like `config.env.bar=foo`:
```python
    flat_infra_overrides = {}
    for override in overrides or []:
        key, value = override.split("=", 1)
        flat_infra_overrides[tuple(key.split("."))] = value
    infra_overrides = flatdict_to_dict(flat_infra_overrides)
```

Reply:
Awesome!  I added this and added a test to check that the deeply nested dictionary worked.